File: GameLibrary/src/cc/lib/zombicide/ZSpell.java
package cc.lib.zombicide;

import cc.lib.game.Utils;
import cc.lib.utils.Table;

public class ZSpell extends ZEquipment<ZSpellType> {

    static {
        addAllFields(ZSpell.class);
    }

    final ZSpellType type;

    public ZSpell() {
        this(null);
    }

    ZSpell(ZSpellType type) {
        this.type = type;
    }

    @Override
    public ZEquipSlotType getSlotType() {
        return ZEquipSlotType.HAND;
    }

    @Override
    public boolean canEquip() {
        return true;
    }

    @Override
    public boolean isEnchantment() {
        return true;
    }

    @Override
    public ZSpellType getType() {
        return type;
    }

    @Override
    public Table getCardInfo(ZCharacter c, ZGame game) {
        Table t = new Table(type.name()).setNoBorder();
        return t.addRow(Utils.wrapTextWithNewlines(type.description, 20));
    }

}

File: GameLibrary/src/cc/lib/zombicide/ZSpell.kt
package cc.lib.zombicide


import cc.lib.utils.Table
import cc.lib.utils.wrap

class ZSpell(override val type: ZSpellType=ZSpellType.HEALING) : ZEquipment<ZSpellType>() {
    companion object {
        init {
            addAllFields(ZSpell::class.java)
        }
    }

    override val slotType: ZEquipSlotType
        get() = ZEquipSlotType.HAND

    override fun isEquippable(c: ZCharacter): Boolean {
        return true
    }

    override val isEnchantment: Boolean
        get() = true

    override fun getCardInfo(c: ZCharacter, game: ZGame): Table {
        val t = Table(label).setNoBorder()
        return t.addRow(type.description.wrap( 20))
    }
}